htlcswitch/link.go

```diff
@@ -3369,10 +3369,11 @@ func (l *channelLink) processRemoteAdds(fwdPkg *channeldb.FwdPkg,
 				// Otherwise, it was already processed, we can
 				// can collect it and continue.
 				addMsg := &lnwire.UpdateAddHTLC{
-					Expiry:        fwdInfo.OutgoingCTLV,
-					Amount:        fwdInfo.AmountToForward,
-					PaymentHash:   pd.RHash,
-					BlindingPoint: fwdInfo.NextBlinding,
+					Expiry:            fwdInfo.OutgoingCTLV,
+					Amount:            fwdInfo.AmountToForward,
+					PaymentHash:       pd.RHash,
+					BlindingPoint:     fwdInfo.NextBlinding,
+					CustomRecordsBlob: pd.WireRecordsBlob,
```

Comment:
Hmm, I don't think we want to store this here. Instead, we should add it only to the `htlcPacket` field we generate. This'll transport the wire TLV records of the _incoming_ HTLC. But here we've already modified the _outgoing_ HTLC to have these values (the interceptor might wnat them to be different, or they shouldn't propagate to this hop as well).

Reply:
Not sure if you are referring to payment descriptor field

On the above line the interceptor hasn't yet been called, and we populate the custom records as a means for the user to read the incoming wire records. If that is changed while intercepting the field will change value